### PR TITLE
External dependency

### PR DESCRIPTION
**Pull request by LokiMidgard**
Added test for pull request #62

### CODE REVIEW

File: src/CodeGeneration.Roslyn.Tests/CodeGenerationTests.cs [hunk 1]
```diff
@@ -23,16 +23,31 @@ public void SimpleGenerationWorks()
...
     }
 
+    [Fact]
+    public void ExternalDependencyFound()
+    {
+        dynamic d = new Wraper();
+        d.TestMethdoSuffix();
+    }
+
+    public partial class Wraper
```
Comment (AArnott) [inline src/CodeGeneration.Roslyn.Tests/CodeGenerationTests.cs]: Typo: Wrapper

File: src/CodeGeneration.Roslyn.Tests/CodeGenerationTests.cs [hunk 2]
```diff
@@ -23,16 +23,31 @@ public void SimpleGenerationWorks()
...
+    {
+        [ExternalDuplicateWithSuffixByName("Suffix")]
+        public void TestMethdo() { }
```
Comment (AArnott) [inline src/CodeGeneration.Roslyn.Tests/CodeGenerationTests.cs]: Typo: TestMethod

File: src/CodeGeneration.Roslyn/CompilationGenerator.cs [hunk 3]
```diff
@@ -184,7 +184,9 @@ protected virtual Assembly LoadAssembly(string path)
             var loadContext = AssemblyLoadContext.GetLoadContext(this.GetType().GetTypeInfo().Assembly);
             var assembly = loadContext.LoadFromAssemblyPath(path);
 
-            this.dependencyContext = this.dependencyContext.Merge(DependencyContext.Load(assembly));
+            var newDependencyContext = DependencyContext.Load(assembly);
+            if(newDependencyContext!=null)
```
Comment (AArnott) [inline src/CodeGeneration.Roslyn/CompilationGenerator.cs]: Please execute Format Document to fix up whitespace.

Comment (LokiMidgard) [reply src/CodeGeneration.Roslyn/CompilationGenerator.cs]: Sorry I forgot. Should now be formated correctly.

Comment (LokiMidgard) [reply src/CodeGeneration.Roslyn/CompilationGenerator.cs]: I fixed the Naming and Build and test it locally. This time it should Work.

Comment (LokiMidgard) [reply src/CodeGeneration.Roslyn/CompilationGenerator.cs]: By the way for me the xUnit tests are not discoverable (in Visual Studio TestExplorer) untill I update the xunit library and the xunit runner to version 2.3.1.

On the CI it works so I haven't checked those changes in.

*Don't want to break it again*

Comment (AArnott) [reply src/CodeGeneration.Roslyn/CompilationGenerator.cs]: Good to know. I'll send a PR that updates that.

**Comment by amis92** [comment]
Releasing these changes to nuget would be nice, as the current build crashes in all my projects, because there was no support for no-depContext assemblies.
@AArnott

**Comment by AArnott** [comment]
0.4.49 that includes this PR pushing to nuget.org now.

**Pull request merged**